yugabyte/yugabyte-db
# [#7706] [Platform] [Backport-2.6] Edit backup config credentials.

**Description:**
Backup config security credentials can change over time. The platform should support editing the security credentials even if there are existing backups associated with a Backup Config. When security credentials expire, the Platform will fail to take new backups and also fail to delete old backups.

This change should be implemented for all three storage configs:

AWS: Allow editing Access Key/Secret, also toggling between IAM and AccessKey
GCP: Allow editing GCS Credentials [Do not allow editing bucket name]
Azure: Allow editing Azure SAS token [Do not allow editing Container URL]

**Test Plan:**
1. Hit the config button from the SideNav.
2. Click on Backup Tab and then hit the amazon s3 tab.
3. Create a backup config for s3.
4. Once it got created then you'll see the Edit configuration button next to the Delete configuration button.
5. Click on the Edit Configuration button.
6. AWS access credentials should be editable.
7. Do the same for the rest of the configs.

**SBT test:**

```[error] Failed: Total 1291, Failed 2, Errors 0, Passed 1285, Skipped 4
[error] Failed tests:
[error]         com.yugabyte.yw.controllers.SessionControllerTest
[error]         com.yugabyte.yw.commissioner.tasks.EditKubernetesUniverseTest
[error] (Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 1084 s (18:04), completed 21 May, 2021 1:55:49 PM
mahendra@mahendra-HP-ZBook-14u-G5:~/yugabyte/yugabyte-db/managed$ sbt "testOnly com.yugabyte.yw.commissioner.tasks.EditKubernetesUniverseTest"
[info] welcome to sbt 1.3.13 (Private Build Java 1.8.0_292)
[info] loading settings for project managed-build from plugins.sbt ...
[info] loading project definition from /home/mahendra/yugabyte/yugabyte-db/managed/project
[info] loading settings for project root from ui-build.sbt,build.sbt ...
[info] set current project to yugaware (in build file:/home/mahendra/yugabyte/yugabyte-db/managed/)
[Yugabyte sbt log] [Resolver] Maven cache server (such as Nexus or Artifactory), specified by YB_MVN_CACHE_URL: List()
[Yugabyte sbt log] [Resolver] Local resolver (enabled by USE_MAVEN_LOCAL, path can be customized with YB_MVN_LOCAL_REPO): List(cache:Maven2 Local: /home/mahendra/.m2/repository)
[Yugabyte sbt log] [Resolver] Default resolver: Vector(FileRepository(local, Patterns(ivyPatterns=Vector(${ivy.home}/local/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), artifactPatterns=Vector(${ivy.home}/local/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), isMavenCompatible=false, descriptorOptional=false, skipConsistencyCheck=false), FileConfiguration(true, None)), public: https://repo1.maven.org/maven2/)
[Yugabyte sbt log] [Resolver] Snapshot resolver for yb-client jar (used when USE_MAVEN_LOCAL is not set, mostly during local development, configured with YB_MVN_SNAPSHOT_URL): List()
[info] Instrumenting 700 classes to /home/mahendra/yugabyte/yugabyte-db/managed/target/scala-2.12/jacoco/instrumented-classes
[info] Test run started
[info] Test com.yugabyte.yw.commissioner.tasks.EditKubernetesUniverseTest.testAddNode started
[info] Test com.yugabyte.yw.commissioner.tasks.EditKubernetesUniverseTest.testRemoveNode started
[info] Test com.yugabyte.yw.commissioner.tasks.EditKubernetesUniverseTest.testChangeInstanceType started
[info] Test run finished: 0 failed, 0 ignored, 3 total, 54.164s
[info] Passed: Total 3, Failed 0, Errors 0, Passed 3
[success] Total time: 59 s, completed 21 May, 2021 1:58:39 PM
```
com.yugabyte.yw.controllers.SessionControllerTest expected to fail..
